mysql/check.py

```diff
@@ -1008,24 +1014,53 @@ def _get_stats_from_innodb_status(self, db):
                 results['Innodb_os_file_writes'] = long(row[4])
                 results['Innodb_os_file_fsyncs'] = long(row[8])
             elif line.find('Pending normal aio reads:') == 0:
-                # Pending normal aio reads: 0, aio writes: 0,
-                # or Pending normal aio reads: 0 [0, 0] , aio writes: 0 [0, 0] ,
-                # or Pending normal aio reads: [0, 0, 0, 0] , aio writes: [0, 0, 0, 0] ,
-                # or Pending normal aio reads: 0 [0, 0, 0, 0] , aio writes: 0 [0, 0, 0, 0] ,
-                if len(row) == 14:
-                    results['Innodb_pending_normal_aio_reads'] = long(row[4])
-                    results['Innodb_pending_normal_aio_writes'] = long(row[10])
+                if len(row) == 8:
+                    # (len(row) == 8)  Pending normal aio reads: 0, aio writes: 0,
+                    if row[4].isdigit() and row[7].isdigit():
```

Comment:
It's nice that you did this, but I don't think you really needed to ;)

The actual `long(row[n])` will raise a ValueError if we can't parse to a long. So wrapping this entire _Pending normal aio reads_ in a `try...except ValueError` would've probably sufficed. 😅 

This would probably apply to all the blocks below.